### PR TITLE
Add more debugging data for bad http responses

### LINKED ISSUE #636: JSON RPC client: Report that a response returned a bad error code
### Setup

**CometBFT version** ~all

**Have you tried the latest version**: yes

### What happened?

When running a ABCI query, through the standard tendermint http client from: github.com/tendermint/tendermint/rpc/client/http , I will occasionally get errors of the form:
```
error unmarshalling: invalid character '<' looking for beginning of value
```

### What did you expect to happen?

By adding a println in the JSON http client, called within this (https://github.com/cometbft/cometbft/blob/main/rpc/jsonrpc/client/http_json_client.go#L213-L216) , I found out that this is due to the HTTP response being:

```
429 Too Many Requests
```

I would have expected to get an error reflecting the actual issue, rather than this incorrect JSON parsing.

Furthermore, I've seen this error in several other contexts. (E.g. mistakenly using http://  instead of https://)

Fixing this will likely be of significant debugging aid to everyone in cosmos.

### PR DESCRIPTION
**Pull request by ValarDragon**
Closes #636 

No new tests are added, because as far as I can tell, the relevant method here has no go tests.

I have tested this on my node, and it returns error as expected:
```
error in json rpc client, with http response metadata: (Status: 429 Too Many Requests, Protocol HTTP/1.1). error unmarshalling: invalid character '<' looking for beginning of value
```

Would appreciate this getting backported to the various release lines :pray: 

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by thanethomson — COMMENTED** [review]
Seems reasonable to me! Just one minor issue re: the changelog entry.

So you'd like this to be backported all the way back to the v0.34 series?

File: .changelog/v0.34.27/improvements/638-json-rpc-error-message.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- `[jsonrpc/client]` Improve the error message for client errors stemming from
```
[the remark this answers is not in the document]
Comment (ValarDragon) [reply .changelog/v0.34.27/improvements/638-json-rpc-error-message.md:1]: oh thank you! Could you amend the PR template, this was unclear to me as a newcomer!

[the remark this answers is not in the document]
Comment (thanethomson) [reply .changelog/v0.34.27/improvements/638-json-rpc-error-message.md]: Interesting! `unclog` usually does this for you automatically when you add a new entry using the CLI. But I'll make a note and amend the PR template :+1:

**Review by thanethomson — APPROVED** [review]

**Comment by thanethomson** [comment]
Oh @ValarDragon, could you please update with the latest changes from `main`, or allow us to do so to your branch?

**Pull request merged**